### PR TITLE
Add CI tasks for docker images

### PR DESCRIPTION
**Pull request by cpennington**
@e0d @feanil: I'd love some review on this.

### CODE REVIEW

**Comment by cpennington** [comment]
Also, note that this will only run once #2511 has been merged.

**Comment by cpennington** [comment]
@mulby: Too many `Makefile` shenanigans, or just the right amount?

File: .travis.yml [hunk 1]
```diff
@@ -1,63 +1,34 @@
...
+
 branches:
   only:
     - master
+
 python:
   - "2.7"
+
+services:
+  - docker
```
Comment (e0d) [inline .travis.yml:12]: Can we specify minimum version here?  Is that even an option?

File: .travis.yml [hunk 2]
```diff
@@ -1,63 +1,34 @@
...
+  
+addons:
+  apt:
+    packages:
+      - nodejs
```
Comment (e0d) [inline .travis.yml:17]: Why is this here?

File: Makefile [hunk 3]
```diff
@@ -1,14 +1,43 @@
...
+	@echo '    make test                      run all tests'
+	@echo '    make build                     build everything'
+	@echo '    make pkg                       package everything'
+	@echo '    make clean                     remove build by-products'
+	@echo ''
+	@echo '    Docker:'
+	@echo '        $$image: any dockerhub image'
+	@echo '        $$container: any container defined in docker/build/$$container/Dockerfile'
+	@echo ''
+	@echo '        make $(docker_pull)$$image        pull $$image from dockerhub'
```
Comment (e0d) [inline Makefile:21]: Why is this needed a make provided command?  Are we providing a competing idiom?

File: docker/build/elasticsearch/xDockerfile [hunk 4]
```diff
@@ -1,7 +1,6 @@
 FROM edxops/precise-common:latest
```
Comment (e0d) [inline docker/build/elasticsearch/xDockerfile:1]: What's up with all these renames?

**Comment by e0d** [comment]
@cpennington Lot's of good stuff here, but have some questions.  If the Dockerfile were renamed to disable them, I'd like to address whatever issues necessitated that before merging.

**Comment by cpennington** [comment]
@e0d: Those files are broken on master. I'd rather get the CI integration merged, and then make separate PRs to fix the disabled files individually.

**Comment by e0d** [comment]
@cpennington I'm OK with that.  I had some other questions as well, but nothing too big.

File: docker/build/elasticsearch/xDockerfile (hunk 4 above)
Comment (cpennington) [inline docker/build/elasticsearch/xDockerfile:1]: As you surmised, these are disabling dockerfiles.

File: Makefile (hunk 3 above)
Comment (cpennington) [inline Makefile:21]: It's used so that we can use it as a pre-requisite task and not pull multiple times.

File: .travis.yml (hunk 1 above)
Comment (cpennington) [inline .travis.yml:12]: Sadly, no.

File: .travis.yml (hunk 2 above)
Comment (cpennington) [inline .travis.yml:17]: Because it was install using `sudo apt-get install` below.

**Comment by cpennington** [comment]
@e0d: Good to merge?

**Comment by e0d** [comment]
:+1:
